.github/workflows/dashboard-testnet.yml

```diff
@@ -102,41 +102,26 @@ jobs:
               @keep-network/keep-core@${{ steps.upstream-builds-query.outputs.keep-core-contracts-version }} \
               @keep-network/keep-ecdsa@${{ steps.upstream-builds-query.outputs.keep-ecdsa-contracts-version }} \
               @keep-network/tbtc@${{ steps.upstream-builds-query.outputs.tbtc-contracts-version }} \
-              @keep-network/coverage-pools@${{ steps.upstream-builds-query.outputs.coverage-pools-version }}
+              @keep-network/coverage-pools@0.0.1-dev.0
+#              @keep-network/coverage-pools@${{ steps.upstream-builds-query.outputs.coverage-pools-version }}
 
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+      - name: NPM install
+        run: npm ci
 
-      - name: Cache Docker layers
-        uses: actions/cache@v2
-        with:
-          path: /tmp/.buildx-cache
-          key: ${{ runner.os }}-buildx-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-buildx-
+      - name: NPM build
+        run: npm run build
 
-      - name: Login to Google Container Registry
+      - name: Deploy to GCP bucket
         if: github.event_name == 'workflow_dispatch'
-        uses: docker/login-action@v1
-        with:
-          registry: ${{ env.GCR_REGISTRY_URL }}
-          username: _json_key
-          password: ${{ secrets.KEEP_TEST_GCR_JSON_KEY }}
-
-      - name: Build and publish Keep Token Dashboard image
-        uses: docker/build-push-action@v2
-        env:
-          IMAGE_NAME: 'keep-dapp-token-dashboard'
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
         with:
-          context: ./solidity/dashboard/
-          # GCR image should be named according to following convention:
-          # HOSTNAME/PROJECT-ID/IMAGE:TAG
-          # We don't use TAG yet, will be added at later stages of work on RFC-18.
-          tags: ${{ env.GCR_REGISTRY_URL }}/${{ env.GOOGLE_PROJECT_ID }}/${{ env.IMAGE_NAME }}
-          labels: revision=${{ github.sha }}
-          push: ${{ github.event_name == 'workflow_dispatch' }}
-          cache-from: type=local,src=/tmp/.buildx-cache
-          cache-to: type=local,dest=/tmp/.buildx-cache-new
+          service-key: ${{ secrets.KEEP_TEST_CI_UPLOAD_DAPP_JSON_KEY_BASE64 }}
+          project: ${{ env.GOOGLE_PROJECT_ID }}
+          bucket-name: keep-test-dashboard-dapp
```

Comment:
We've generally been using a bucket name that matches the intended domain name here so the DNS setup can just CNAME to `c.storage.googleapis.com` (e.g. `preview.keep.network` as the bucket name, `preview.keep.network` CNAME `c.storage.googleapis.com` as the DNS record). I realize there's a possibility we may not strictly need this approach here since the testnet.keep.network DNS management is delegated to GCP, but it may still makes sense to stick to this pattern for consistency with the top-level domains and other deployments that use this strategy.

Will leave the ultimate decision up to you---just wanted to mention it before we finalized things here.

Reply:
Yep I'm aware but if we want to use HTTPS we need to use a load balancer (https://cloud.google.com/storage/docs/hosting-static-website#lb-ssl). In that case, the bucket name doesn't matter as we set A records to the LB external IP.

Reply:
Here it is https://github.com/keep-network/keep-core/pull/2562/commits/adcba5043b748a0299aa705eb593e045a85d2ee3